### ISSUE TITLE
bad interaction with SpecialFunctions

### ISSUE THREAD

**Issue by CarloLucibello** [JuliaGPU/CuArrays.jl#56]
On julia 0.6
```julia
julia> using CuArrays

julia> x=cu(rand(3))
3-element CuArray{Float32,1}:
 0.130609
 0.036018
 0.938617

julia> using SpecialFunctions

julia> erf.(x)
3-element CuArray{Float32,1}:
Error showing value of type CuArray{Float32,1}:
ERROR: CUDA error: unspecified launch failure (code #719, ERROR_LAUNCH_FAILED)
Stacktrace:
 [1] macro expansion at /home/lucibello/.julia/v0.6/CUDAdrv/src/base.jl:148 [inlined]
 [2] #download!#5(::Bool, ::Function, ::Base.RefArray{Float32,Array{Float32,1},Void}, ::CUDAdrv.Mem.Buffer, ::Int64, ::CUDAdrv.CuStream) at /home/lucibello/.julia/v0.6/CUDAdrv/src/memory.jl:224
 [3] (::CUDAdrv.Mem.#kw##download!)(::Array{Any,1}, ::CUDAdrv.Mem.#download!, ::Base.RefArray{Float32,Array{Float32,1},Void}, ::CUDAdrv.Mem.Buffer, ::Int64, ::CUDAdrv.CuStream) at ./<missing>:0
 [4] #download!#8 at /home/lucibello/.julia/v0.6/CUDAdrv/src/memory.jl:292 [inlined]
 [5] download! at /home/lucibello/.julia/v0.6/CUDAdrv/src/memory.jl:291 [inlined] (repeats 2 times)
 [6] copy!(::Array{Float32,1}, ::CuArray{Float32,1}) at /home/lucibello/.julia/v0.6/CuArrays/src/array.jl:66
 [7] #showarray#3(::Bool, ::Function, ::IOContext{Base.Terminals.TTYTerminal}, ::CuArray{Float32,1}, ::Bool) at /home/lucibello/.julia/v0.6/CuArrays/src/array.jl:143
 [8] display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::MIME{Symbol("text/plain")}, ::CuArray{Float32,1}) at ./REPL.jl:122
 [9] display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::CuArray{Float32,1}) at ./REPL.jl:125
 [10] display(::CuArray{Float32,1}) at ./multimedia.jl:194
 [11] eval(::Module, ::Any) at ./boot.jl:235
 [12] print_response(::Base.Terminals.TTYTerminal, ::Any, ::Void, ::Bool, ::Bool, ::Void) at ./REPL.jl:144
 [13] print_response(::Base.REPL.LineEditREPL, ::Any, ::Void, ::Bool, ::Bool) at ./REPL.jl:129
 [14] (::Base.REPL.#do_respond#16{Bool,Base.REPL.##26#36{Base.REPL.LineEditREPL,Base.REPL.REPLHistoryProvider},Base.REPL.LineEditREPL,Base.LineEdit.Prompt})(::Base.LineEdit.MIState, ::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Bool) at ./REPL.jl:646

julia> x
3-element CuArray{Float32,1}:
Error showing value of type CuArray{Float32,1}:
ERROR: CUDA error: unspecified launch failure (code #719, ERROR_LAUNCH_FAILED)
Stacktrace:
 [1] macro expansion at /home/lucibello/.julia/v0.6/CUDAdrv/src/base.jl:148 [inlined]
 [2] #download!#5(::Bool, ::Function, ::Base.RefArray{Float32,Array{Float32,1},Void}, ::CUDAdrv.Mem.Buffer, ::Int64, ::CUDAdrv.CuStream) at /home/lucibello/.julia/v0.6/CUDAdrv/src/memory.jl:224
 [3] (::CUDAdrv.Mem.#kw##download!)(::Array{Any,1}, ::CUDAdrv.Mem.#download!, ::Base.RefArray{Float32,Array{Float32,1},Void}, ::CUDAdrv.Mem.Buffer, ::Int64, ::CUDAdrv.CuStream) at ./<missing>:0
 [4] #download!#8 at /home/lucibello/.julia/v0.6/CUDAdrv/src/memory.jl:292 [inlined]
 [5] download! at /home/lucibello/.julia/v0.6/CUDAdrv/src/memory.jl:291 [inlined] (repeats 2 times)
 [6] copy!(::Array{Float32,1}, ::CuArray{Float32,1}) at /home/lucibello/.julia/v0.6/CuArrays/src/array.jl:66
 [7] #showarray#3(::Bool, ::Function, ::IOContext{Base.Terminals.TTYTerminal}, ::CuArray{Float32,1}, ::Bool) at /home/lucibello/.julia/v0.6/CuArrays/src/array.jl:143
 [8] display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::MIME{Symbol("text/plain")}, ::CuArray{Float32,1}) at ./REPL.jl:122
 [9] display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::CuArray{Float32,1}) at ./REPL.jl:125
 [10] display(::CuArray{Float32,1}) at ./multimedia.jl:194
 [11] eval(::Module, ::Any) at ./boot.jl:235
 [12] print_response(::Base.Terminals.TTYTerminal, ::Any, ::Void, ::Bool, ::Bool, ::Void) at ./REPL.jl:144
 [13] print_response(::Base.REPL.LineEditREPL, ::Any, ::Void, ::Bool, ::Bool) at ./REPL.jl:129
 [14] (::Base.REPL.#do_respond#16{Bool,Base.REPL.##26#36{Base.REPL.LineEditREPL,Base.REPL.REPLHistoryProvider},Base.REPL.LineEditREPL,Base.LineEdit.Prompt})(::Base.LineEdit.MIState, ::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Bool) at ./REPL.jl:646

**Comment by CarloLucibello** [comment]
note: the same happens without calling `using SpecialFunctions`, so in both cases this may be due to the deprecation in Base

**Comment by CarloLucibello** [comment]
wait, I think it is a problem with my installation
```julia
julia> cos.(x)
3-element CuArray{Float32,1}:
Error showing value of type CuArray{Float32,1}:
ERROR: CUDA error: unspecified launch failure (code #719, ERROR_LAUNCH_FAILED)
Stacktrace:
 [1] macro expansion at /home/lucibello/.julia/v0.6/CUDAdrv/src/base.jl:148 [inlined]
 [2] #download!#5(::Bool, ::Function, ::Base.RefArray{Float32,Array{Float32,1},Void}, ::CUDAdrv.Mem.Buffer, ::Int64, ::CUDAdrv.CuStream) at /home/lucibello/.julia/v0.6/CUDAdrv/src/memory.jl:224
 [3] (::CUDAdrv.Mem.#kw##download!)(::Array{Any,1}, ::CUDAdrv.Mem.#download!, ::Base.RefArray{Float32,Array{Float32,1},Void}, ::CUDAdrv.Mem.Buffer, ::Int64, ::CUDAdrv.CuStream) at ./<missing>:0
 [4] #download!#8 at /home/lucibello/.julia/v0.6/CUDAdrv/src/memory.jl:292 [inlined]
 [5] download! at /home/lucibello/.julia/v0.6/CUDAdrv/src/memory.jl:291 [inlined] (repeats 2 times)
 [6] copy!(::Array{Float32,1}, ::CuArray{Float32,1}) at /home/lucibello/.julia/v0.6/CuArrays/src/array.jl:66
 [7] #showarray#3(::Bool, ::Function, ::IOContext{Base.Terminals.TTYTerminal}, ::CuArray{Float32,1}, ::Bool) at /home/lucibello/.julia/v0.6/CuArrays/src/array.jl:143
 [8] display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::MIME{Symbol("text/plain")}, ::CuArray{Float32,1}) at ./REPL.jl:122
 [9] display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::CuArray{Float32,1}) at ./REPL.jl:125
 [10] display(::CuArray{Float32,1}) at ./multimedia.jl:194
 [11] eval(::Module, ::Any) at ./boot.jl:235
 [12] print_response(::Base.Terminals.TTYTerminal, ::Any, ::Void, ::Bool, ::Bool, ::Void) at ./REPL.jl:144
 [13] print_response(::Base.REPL.LineEditREPL, ::Any, ::Void, ::Bool, ::Bool) at ./REPL.jl:129
 [14] (::Base.REPL.#do_respond#16{Bool,Base.REPL.##26#36{Base.REPL.LineEditREPL,Base.REPL.REPLHistoryProvider},Base.REPL.LineEditREPL,Base.LineEdit.Prompt})(::Base.LineEdit.MIState, ::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Bool) at ./REPL.jl:646

```

**Comment by maleadt** [comment]
This is just a case of Base intrinsics not being supported by CuArrays. Use CUDAnative intrinsics instead.
Also, the `ERROR_LAUNCH_FAILED` is sticky. You need to restart your Julia session after encountering one.

**Comment by CarloLucibello** [comment]
Ok, thanks! 
```julia
julia> CUDAnative.cos.(x)
3-element CuArray{Float32,1}:
 0.978247
 0.975785
 0.833308

julia> CUDAnative.erf.(x)
3-element CuArray{Float32,1}:
 0.232402
 0.244851
 0.592528
```
I'll wait julia 0.7 for a more comfortable experience

**Comment by maleadt** [comment]
> I'll wait julia 0.7 for a more comfortable experience

Sadly, this particular issue isn't fixed on 0.7, due to issues with the Julia compiler that are only planned to be fixed in 1.1.

**Comment by MikeInnes** [comment]
Worth looking at how [Flux](https://github.com/FluxML/Flux.jl/blob/9ccbac8b80cc3dc50f0fdbf02063d811add7b539/src/layers/basic.jl#L78) handles this.
